src/java.base/share/classes/java/lang/invoke/GenerateJLIClassesHelper.java

```diff
@@ -176,9 +184,12 @@ Map<String, byte[]> build() {
             result.put(BASIC_FORMS_HOLDER,
                        generateBasicFormsClassBytes(BASIC_FORMS_HOLDER));
 
-            speciesTypes.forEach(types -> {
-                Map.Entry<String, byte[]> entry = generateConcreteBMHClassBytes(types);
-                result.put(entry.getKey(), entry.getValue());
+            speciesTypes.forEach(new Consumer<String>() {
+                @Override
+                public void accept(String types) {
+                    Map.Entry<String, byte[]> entry = generateConcreteBMHClassBytes(types);
+                    result.put(entry.getKey(), entry.getValue());
+                }
             });
```

Comment:
```suggestion
            speciesTypes.forEach(types -> {
                Map.Entry<String, byte[]> entry = generateConcreteBMHClassBytes(types);
                result.put(entry.getKey(), entry.getValue());
            });
```

Reply:
Unfortunately JDK bootstrap library generating lambdas cannot use lambdas.
Suggested patch would lead to stack overflow during JDK bootstrap.

Reply:
According to the diff here this previously was a lambda though, wondering what changed to make it fail now.

Reply:
`GenerateJLIClassesHelper` is only called by the **jlink** `GenerateJLIClassesPlugin`, and **jlink** only runs after the **JVM** bootstrap has finished, so there shouldn’t be any cycle here.